cmd/thanos/store.go

```diff
@@ -84,6 +84,11 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application) {
 	consistencyDelay := modelDuration(cmd.Flag("consistency-delay", "Minimum age of all blocks before they are being read. Set it to safe value (e.g 30m) if your object storage is eventually consistent. GCS and S3 are (roughly) strongly consistent.").
 		Default("0s"))
 
+	ignoreDeletionMarksDelay := modelDuration(cmd.Flag("ignore-deletion-marks-delay", "Duration after which the blocks marked for deletion can be filtered out. "+
+		"If delete-delay duration provided to compactor or bucket verify component, it will upload deletion-mark.json file to mark after what duration the block should be deleted rather than deleting the block straight away. "+
+		"If delete-delay is non-zero for compactor or bucket verify component, ignore-deletion-marks-delay should be set to (delete-delay)/2 so that blocks marked for deletion are filtered out.").
```

Comment:
```suggestion
		"If delete-delay is non-zero for compactor or bucket verify component, ignore-deletion-marks-delay should be set to (delete-delay)/2 so that blocks marked for deletion are filtered out before being deleted from bucket").
```

Reply:
I have added some more explanation in the first line as well, changed from `filtered out` to `filtered out while fetching blocks`, that makes it more clear according to me.

Reply:
I am resolving this thread for now, if there are feedback points, feel free to open this again or make new thread. Thanks.